src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallManager.cs

```diff
@@ -0,0 +1,100 @@
+        private readonly ReleaseVersion _sdkVersion;
+
+        public WorkloadInstallManager(
+            IReporter reporter,
+            IWorkloadInstaller workloadInstaller,
+            IWorkloadResolver workloadResolver,
+            string version = null)
+        {
+            _sdkVersion = new ReleaseVersion(version ?? Product.Version);
+            _workloadInstaller = workloadInstaller;
+            _workloadResolver = workloadResolver;
+            _reporter = reporter;
+        }
+
+        public void InstallWorkloads(IEnumerable<WorkloadId> workloadIds, bool skipManifestUpdate = false)
+        {
+            _reporter.WriteLine();
+            var featureBand = new SdkFeatureBand(string.Join('.', _sdkVersion.Major, _sdkVersion.Minor, _sdkVersion.SdkFeatureBand));
+
+            if (!skipManifestUpdate)
+            {
+                throw new NotImplementedException();
+            }
+
+            InstallWorkloadsWithInstallRecord(workloadIds, featureBand);
+
+            if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+            {
+                (_workloadInstaller as PackWorkloadInstallerBase).GarbageCollectInstalledWorkloadPacks();
+            }
+
+            _reporter.WriteLine();
+            _reporter.WriteLine(string.Format(LocalizableStrings.InstallationSucceeded, string.Join(", ", workloadIds)));
+            _reporter.WriteLine();
+        }
+
+        private void InstallWorkloadsWithInstallRecord(IEnumerable<WorkloadId> workloadIds, SdkFeatureBand sdkFeatureBand)
+        {
+            if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+            {
+                var installer = _workloadInstaller as PackWorkloadInstallerBase;
+
+                var workloadPacksMap = workloadIds
+                    .Select(workloadId => (workloadId, _workloadResolver.GetPacksInWorkload(workloadId.ToString()).Select(packId => _workloadResolver.TryGetPackInfo(packId))));
+                TransactionalAction.Run(
+                    action: () =>
+                    {
+                        foreach ((var workloadId, var packsToInstall) in workloadPacksMap)
+                        {
+                            foreach (var packId in packsToInstall)
+                            {
+                                installer.InstallWorkloadPack(packId, sdkFeatureBand);
+                            }
+
+                            _workloadInstaller.WriteWorkloadInstallationRecord(workloadId, sdkFeatureBand);
+                        }
+
+                    },
+                    rollback: () => {
+                        foreach ((var workloadId, var packsToInstall) in workloadPacksMap)
+                        {
+                            foreach (var packId in packsToInstall)
+                            {
+                                installer.RollBackWorkloadPackInstall(packId, sdkFeatureBand);
```

Comment:
I think there's an issue with this rollback logic.  A workload pack might already have been installed for another workload that was already installed, but it looks like if we were to roll back the install of a workload that also depended on the workload pack, we'd end up deleting the workload pack and breaking the previously installed workload.

Reply:
I noticed this bug in too and changed it in my latest commit, now I'm checking to make sure that there aren't any pack install records left before uninstalling the pack.